app/src/main/java/tech/ula/ui/AppListAdapter.kt

```diff
@@ -47,92 +65,177 @@ class AppListAdapter(
             categoriesAndAppsWithDistributionsFirst[firstDisplayCategory] =
                     categoriesAndApps.remove(firstDisplayCategory)!!
         }
-        categoriesAndAppsWithDistributionsFirst.putAll(categoriesAndApps)
+
+        val sortedCategoriesAndApps = categoriesAndApps.toSortedMap()
+        categoriesAndAppsWithDistributionsFirst.putAll(sortedCategoriesAndApps)
 
         categoriesAndAppsWithDistributionsFirst.forEach {
             (category, categoryApps) ->
             val categoryWithPaymentInformation = category + " " +
                     if (category == firstDisplayCategory) freeAnnotation else paidAnnotation
             listBuilder.add(AppSeparatorItem(categoryWithPaymentInformation.capitalize()))
-            categoryApps.forEach { listBuilder.add(AppItem(it)) }
+
+            val sortedCategoryApps = categoryApps.sortedWith(compareBy { it.name })
+            sortedCategoryApps.forEach { listBuilder.add(AppItem(it)) }
         }
-        listBuilder.toList()
+
+        return listBuilder
     }
 
-    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
-        val view: View?
-        val viewHolder: ViewHolder
+    fun updateAppsAndSessions(newApps: List<App>, newActiveSessions: List<Session>) {
 
-        val item = appsAndSeparators[position]
+        val newAppsListItems = createAppsItemListWithSeparators(newApps)
 
-        if (convertView == null) {
-            val inflater = activity.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
-            view = when (item) {
-                is AppSeparatorItem -> inflater.inflate(R.layout.list_item_separator, parent, false)
-                is AppItem -> inflater.inflate(R.layout.list_item_app, parent, false)
-            }
+        val diffCallback = AppsItemDiffCallBack(appsAndSeparators, newAppsListItems)
+        val diffResult = DiffUtil.calculateDiff(diffCallback)
+        apps.clear()
+        apps.addAll(newApps)
```

Comment:
Are these two lines still necessary?

Reply:
`apps.clear() and apps.addAll(newApps)` that is

Reply:
It doesn't look like we're using `apps` at all, it looks safe to remove that list along with these two lines